content/en/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes.md

```diff
@@ -107,89 +99,123 @@ Once you have a Linux-based Kubernetes control-plane node you are ready to choos
 
 {{< note >}}
-All code snippets in Windows sections are to be run in a PowerShell environment
-with elevated permissions (Administrator) on the Windows worker node.
+The following instructions are the manual steps to configure nodes. You can use the [image-builder](https://image-builder.sigs.k8s.io/capi/windows/windows.html) used as part of the Cluster Api project implements the following along with other optimizations.
 {{< /note >}}
 
-{{< tabs name="tab-windows-kubeadm-runtime-installation" >}}
+#### Install containerD
 
-{{% tab name="CRI-containerD" %}}
+Follow the instructions for [installing the Containerd runtime on Windows using powershell](../../../setup/production-environment/container-runtimes.md#containerd)
 
-#### Install containerD
+#### Install kubelet
 
-```powershell
-curl.exe -LO https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/Install-Containerd.ps1
-.\Install-Containerd.ps1
+Kubeadm sets some flags via a environment file.  To support this it is common to use [nssm](https://nssm.cc/) to configure kubelet.  To install nssm do the following:
+
+``` powershell
+mkdir c:\k
+$arch = "win64"
+curl.exe -L https://k8stestinfrabinaries.blob.core.windows.net/nssm-mirror/nssm-2.24.zip -o nssm.zip
+tar.exe C c:\k\ -xvf .\nssm.zip --strip-components 2 */$arch/*.exe
 ```
 
-{{< note >}}
-To install a specific version of containerD specify the version with -ContainerDVersion.
+Next we will set `$KubernetesVersion` to the desired version (ex: `$KubernetesVersion="v1.24.0"`), and then run the following commands to install kubelet:
+
+``` powerShell
+curl.exe -L https://dl.k8s.io/$KubernetesVersion/bin/windows/amd64/kubelet.exe -o c:\k\kubelet.exe
+
+@"
+# Start file needed to support kubeadm extra args
+`$FileContent = Get-Content -Path "/var/lib/kubelet/kubeadm-flags.env"
+`$kubeAdmArgs = `$FileContent.TrimStart(`'KUBELET_KUBEADM_ARGS=`').Trim(`'"`')
+
+`$args = "--cert-dir=`$env:SYSTEMDRIVE/var/lib/kubelet/pki",
+        "--config=`$env:SYSTEMDRIVE/var/lib/kubelet/config.yaml",
+        "--bootstrap-kubeconfig=`$env:SYSTEMDRIVE/etc/kubernetes/bootstrap-kubelet.conf",
+        "--kubeconfig=`$env:SYSTEMDRIVE/etc/kubernetes/kubelet.conf",
+        "--hostname-override=$(hostname)",
+        "--enable-debugging-handlers",
+        "--cgroups-per-qos=false",
+        "--enforce-node-allocatable=``"``"",
+        "--resolv-conf=``"``""
+
+`$kubeletCommandLine = "c:\k\kubelet.exe " + (`$args -join " ") + " `$kubeAdmArgs"
+Invoke-Expression `$kubeletCommandLine
+"@ | Set-Content -Path c:\k\Start-kubelet.ps1
+
+c:\k\nssm.exe install kubelet Powershell -ExecutionPolicy Bypass -NoProfile c:\k\Start-kubelet.ps1
+c:\k\nssm.exe set Kubelet AppStdout C:\k\kubelet.log
+c:\k\nssm.exe set Kubelet AppStderr C:\k\kubelet.err.log 
+```
 
-```powershell
-# Example
-.\Install-Containerd.ps1 -ContainerDVersion 1.4.1
+We need to open a firewall port:
```

Comment:
- Please [address the reader as you](https://kubernetes.io/docs/contribute/style/style-guide/#address-the-reader-as-you).
- Allow TCP traffic to the kubelet from the API server
   - Allowing traffic from anywhere might not be as secure as readers imagine; the page should explain how to restrict that traffic to the API server hosts only, or caution the reader about the risks this implies.